model/app.model.js

```diff
@@ -66,3 +66,43 @@ exports.fetchArticlesWithCommentCount = () => {
       return articlesWithFormattedCount;
     });
 };
+
+exports.insertComment = (article_id, username, body, mybody) => {
+  const validKeys = ["username", "body"];
+
+  for (let key in mybody) {
+    if (!validKeys.includes(key)) {
+      return Promise.reject({
+        status: 404,
+        msg: "Bad request",
+      });
+    }
+  }
+
+  return db
+    .query(`SELECT * FROM articles WHERE article_id =$1`, [article_id])
+    .then(({ rowCount }) => {
+      if (rowCount === 0) {
+        return Promise.reject({ status: 404, msg: "Invalid article_id" });
+      }
+    })
+    .then(() => {
+      return db
+        .query(`SELECT * FROM users WHERE username = $1`, [username])
```

Comment:
We dont need to do these 2 queries. The foreign key constraints on the comments table will reject the SQL if the username doesnt exist in the user table, and the article_id if the article id doesnt exist in the articles table.